cse-valid-adapter-app/src/main/java/com/farao_community/farao/cse_valid/adapter/app/service/CseValidAdapterService.java

```diff
@@ -0,0 +1,100 @@
+import com.farao_community.farao.cse_valid.api.resource.CseValidFileResource;
+import com.farao_community.farao.cse_valid.api.resource.CseValidRequest;
+import com.farao_community.farao.gridcapa.task_manager.api.ProcessFileDto;
+import com.farao_community.farao.gridcapa.task_manager.api.TaskDto;
+import com.farao_community.farao.gridcapa.task_manager.api.TaskStatus;
+import com.farao_community.farao.gridcapa.task_manager.api.TaskStatusUpdate;
+import com.farao_community.farao.gridcapa_cse_valid.starter.CseValidClient;
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
+ */
+@Service
+public class CseValidAdapterService {
+
+    private static final String TTC_ADJUSTMENT_FILE_TYPE = "TTC_ADJUSTMENT";
+    private static final String CRAC_FILE_TYPE = "CRAC";
+    private static final String CGM_FILE_TYPE = "CGM";
+    private static final String GLSK_FILE_TYPE = "GLSK";
+    private static final String TASK_STATUS_UPDATE = "task-status-update";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CseValidAdapterService.class);
+    private final CseValidClient cseValidClient;
+    private final CseValidAdapterConfiguration cseValidAdapterConfiguration;
+    private final StreamBridge streamBridge;
+
+    public CseValidAdapterService(CseValidClient cseValidClient, CseValidAdapterConfiguration cseValidAdapterConfiguration, StreamBridge streamBridge) {
+        this.cseValidClient = cseValidClient;
+        this.cseValidAdapterConfiguration = cseValidAdapterConfiguration;
+        this.streamBridge = streamBridge;
+    }
+
+    public void runAsync(TaskDto taskDto) {
+        MDC.put("gridcapa-task-id", taskDto.getId().toString());
+        try {
+            if (taskDto.getStatus() == TaskStatus.READY || taskDto.getStatus() == TaskStatus.SUCCESS || taskDto.getStatus() == TaskStatus.ERROR) {
+                LOGGER.info("Handling manual run request on TS {} ", taskDto.getTimestamp());
+                CseValidRequest cseValidRequest = buildCseValidRequest(taskDto);
+                CompletableFuture.runAsync(() -> cseValidClient.run(cseValidRequest));
+            } else {
+                LOGGER.warn("Failed to handle manual run request on timestamp {} because it is not ready yet", taskDto.getTimestamp());
+            }
+        } catch (Exception e) {
+            LOGGER.error(String.format("Unexpected error occurred during building the request, task %s will not be run. Reason: %s", taskDto.getId().toString(), e.getMessage()));
```

Comment:
You can use a businessLogger for the error so that the final user of the application know the reason of this failure.